app/src/main/java/com/nervousfish/nervousfish/activities/LoginActivity.java

```diff
@@ -42,18 +53,57 @@ protected void onCreate(final Bundle savedInstanceState) {
             LOGGER.error("Failed to retrieve password from database", e);
         }
 
+        final Keyboard keyboard = new Keyboard(this, R.xml.qwerty);
+        this.keyboardView = (KeyboardView) this.findViewById(R.id.keyboardview);
+        this.keyboardView.setKeyboard(keyboard);
+        this.keyboardView.setPreviewEnabled(false);
+        this.keyboardView.setOnKeyboardActionListener(new OnCustomKeyboardActionListener());
+
+        final EditText editPassword = (EditText) this.findViewById(R.id.login_password_input);
+
+        editPassword.setOnFocusChangeListener((v, hasFocus) -> {
+            if (hasFocus) {
+                this.showCustomKeyboard(v);
+            } else {
+                this.hideCustomKeyboard();
+            }
+        });
+        editPassword.setOnClickListener(this::showCustomKeyboard);
+        editPassword.setCustomSelectionActionModeCallback(new LoginActivity.EditPasswordSelectionCallback());
```

Comment:
What do we need this for if the `EditPasswordSelectionCallback` class returns false for all the methods it implements?
 Could you elaborate a bit on what the methods in this class do?